src/test/java/org/folio/notes/controller/NoteControllerTest.java

```diff
@@ -295,11 +322,695 @@ void return404OnDeleteByIdWhenItNotExist() throws Exception {
+      + "?status=ASSIGNED&orderBy=title");
+    var notes = OBJECT_MAPPER.readValue(content, NoteCollection.class).getNotes();
+
+    assertEquals(2, notes.size());
+    assertEquals(firstNote.getTitle(), notes.get(0).getTitle());
+  }
+
+  @Test
+  @DisplayName("Should return list of assigned notes searched and sorted by title order desc")
+  void shouldReturnListOfAssignedNotesSearchedAndSortedByTitleOrderDesc() throws Exception {
+    var firstNote = generateNote().title("Title ABC");
+    var secondNote = generateNote().title("Title ZZZ ABC");
+    var thirdNote = generateNote().title("Title BBB");
+
+    mockMvc.perform(postNote(firstNote)).andExpect(status().isCreated());
+    mockMvc.perform(postNote(secondNote)).andExpect(status().isCreated());
+    mockMvc.perform(postNote(thirdNote)).andExpect(status().isCreated());
+
+    createLinks(firstNote.getId());
+    createLinks(secondNote.getId());
+
+    var content = getNoteLinks("/note-links/domain/" + DOMAIN + "/type/" + PACKAGE_TYPE + "/id/" + PACKAGE_ID_1
+      + "?status=ASSIGNED&orderBy=title&order=desc");
+    var notes = OBJECT_MAPPER.readValue(content, NoteCollection.class).getNotes();
+
+    assertEquals(2, notes.size());
+    assertEquals(secondNote.getTitle(), notes.get(0).getTitle());
+  }
+
+  @Test
+  @DisplayName("Should return list of notes with unassigned status")
+  void shouldReturnListOfNotesWithUnassignedStatus() throws Exception {
+    var firsNoteWithAssignedLink = generateNote();
+    generateNote();
+    createLinks(firsNoteWithAssignedLink.getId());
+
+    var content = getNoteLinks("/note-links/domain/" + DOMAIN + "/type/" + PACKAGE_TYPE + "/id/" + PACKAGE_ID_1
+      + "?status=UNASSIGNED");
+    var notes = OBJECT_MAPPER.readValue(content, NoteCollection.class).getNotes();
+
+    assertEquals(1, notes.size());
+  }
+
+  @Test
+  @DisplayName("Should return list of notes with non existing domain")
+  void shouldReturnListOfNotesWithNonExistingDomain() throws Exception {
+    var firsNoteWithAssignedLink = generateNote();
+    var secondNoteWithUnassignedLink = generateNote();
+
+    var content = getNoteLinks("/note-links/domain/" + NON_EXISTING_DOMAIN + "/type/" + PACKAGE_TYPE + "/id/" + PACKAGE_ID_1
+      + "?status=UNASSIGNED");
+    var notes = OBJECT_MAPPER.readValue(content, NoteCollection.class).getNotes();
+
+    assertEquals(0, notes.size());
+    assertEquals(DEFAULT_LINK_AMOUNT, firsNoteWithAssignedLink.getLinks().size());
+    assertEquals(DEFAULT_LINK_AMOUNT, secondNoteWithUnassignedLink.getLinks().size());
+  }
+
+  @Test
+  @DisplayName("Should return list of notes with status all")
+  void shouldReturnListOfNotesWithStatusAll() throws Exception {
+    generateNote();
+    generateNote();
+
+    var content = getNoteLinks("/note-links/domain/" + DOMAIN + "/type/" + PACKAGE_TYPE + "/id/" + PACKAGE_ID_1 + "?status=all");
+    var notes = OBJECT_MAPPER.readValue(content, NoteCollection.class).getNotes();
+
+    assertEquals(2, notes.size());
```

Comment:
Maybe will be better to add a link to one of the notes to check that all notes are returned: assigned and not assigned?